sup3r/models/surface.py

```diff
@@ -104,34 +119,34 @@ def __len__(self):
         return 1
 
     @classmethod
-    def load(cls, features, s_enhance, verbose=False, **kwargs):
+    def load(cls, model_dir, verbose=False):
         """Load the GAN with its sub-networks from a previously saved-to output
         directory.
 
         Parameters
         ----------
-        features : list
-            List of feature names that this model will operate on for both
-            input and output. This must match the feature axis ordering in the
-            array input to generate(). Typically this is a list containing:
-            temperature_*m, relativehumidity_*m, and pressure_*m. The list can
-            contain multiple instances of each variable at different heights.
-            relativehumidity_*m entries must have corresponding temperature_*m
-            entires at the same hub height.
-        s_enhance : int
-            Integer factor by which the spatial axes are to be enhanced.
+        model_dir : str
+            Directory to load SurfaceSpatialMetModel model files from. Must
+            have a model_params.json file containing "meta" key with all of the
+            class init args.
         verbose : bool
             Flag to log information about the loaded model.
-        kwargs : None | dict
-            Optional kwargs to initialize SurfaceSpatialMetModel
 
         Returns
         -------
         out : SurfaceSpatialMetModel
             Returns an initialized SurfaceSpatialMetModel
         """
 
-        model = cls(features, s_enhance, **kwargs)
+        fp_params = os.path.join(model_dir, 'model_params.json')
+        assert os.path.exists(fp_params), f'Could not find: {fp_params}'
+        with open(fp_params, 'r') as f:
+            params = json.load(f)
+
+        meta = params['meta']
+        args = signature(cls.__init__).parameters
+        kwargs = {k: v for k, v in meta.items() if k in args}
+        model = cls(**kwargs)
```

Comment:
you use this code block in linear.py also, and it seems useful as a separate function or an inherited method?

Reply:
Yeah good point i think we can actually inherit the surface model from the linear model. Let me test it.